containers/ecr-viewer/src/app/components/LibrarySearch.tsx

```diff
@@ -23,24 +25,51 @@ const LibrarySearch = ({
   className,
   textBoxClassName,
 }: LibrarySearchProps) => {
-  const { updateQueryParam, pushQueryUpdate } = useLibraryQueryParam();
+  const [searchTerm, setSearchTerm] = useState<string | undefined>();
+  const prevSearchTermRef = useRef(searchTerm);
```

Comment:
out of curiosity, why a ref here vs a state variable?

Reply:
I don't feel strongly here either way - genuine curiosity as it wasn't what I would have first attempted and I like to learn from other folks' PRs